tool/src/dart/mod.rs

```diff
@@ -544,37 +528,21 @@ impl<'a, 'cx> TyGenContext<'a, 'cx> {
                 SpecialMethod::Iterable => {
                     format!("{return_ty} get iterator")
                 }
+                SpecialMethod::Indexer => {
+                    format!("{return_ty} operator []({params})")
+                }
                 _ => unimplemented!("Found unknown special method type {special:?}"),
             }
-        } else if method.param_self.is_none() {
-            // Static field
-            if params.is_empty()
```

Comment:
I removed this case because I realised that in order to make something a static singleton, it needs to be immutable, and it's hard to figure that out.